src/compiler/utilitiesPublic.ts

```diff
@@ -1046,7 +1046,7 @@ namespace ts {
     // All node tests in the following list should *not* reference parent pointers so that
     // they may be used with transformations.
     /* @internal */
-    export function isNode(node: Node) {
+    export function isNode(node: any): node is Node {

```

Comment:
That's not what this function does. Its a misclassification, unfortunately. This function tests whether a `Node` is "a `Node` that is not a `Token`" (i.e., is has `kind >= SyntaxKind.QualifiedName`). It should probably have been named `isNonTokenNode`, tbh. If you want to check whether something is a `Node`, that is a little trickier. You'd have to check that: 
- the argument is an `object` that is not `null`.
- the argument has `kind`, `pos`, and `end` fields (at a minimum).

Alternatively you'd need to use the *current* value of `objectAllocator` and check whether the value is an `instanceof` an of `objectAllocator.getNodeConstructor()`, `.getTokenConstructor()`, `.getIdentifierConstructor()`, `.getPrivateIdentifierConstructor()`, or `.getSourceFileConstructor()`. You'd have to use the *current* value since the language service overwrites `objectAllocator` with the more complex version.

Either way, I'd recommend a different function for that.

Reply:
Oops, I guess I didn’t check what `isNodeKind` actually does, and didn’t end up needing this change after all anyway. But do note, this function is literally never used.